lib/ansible/modules/cloud/amazon/ec2_metric_alarm.py

```diff
@@ -184,156 +217,173 @@ def create_metric_alarm(connection, module):
+                           'Statistic': statistic,
+                           'ComparisonOperator': comparison,
+                           'Threshold': threshold,
+                           'Period': period,
+                           'EvaluationPeriods': evaluation_periods,
+                           'Unit': unit,
+                           'AlarmDescription': description,
+                           'Dimensions': dimensions,
+                           'TreatMissingData': treat_missing_data}.items():
+            try:
+                if alarm[key] != value:
+                    changed = True
+            except KeyError:
+                if value is not None:
+                    changed = True
+
+            alarm[key] = value
+
+        for key, value in {'AlarmActions': alarm_actions,
+                           'InsufficientDataActions': insufficient_data_actions,
+                           'OKActions': ok_actions}.items():
+            action = value or []
+            if alarm[key] != action:
                 changed = True
-                setattr(alarm, 'dimensions', dim1)
-
-        for attr in ('alarm_actions', 'insufficient_data_actions', 'ok_actions'):
-            action = module.params.get(attr) or []
-            # Boto and/or ansible may provide same elements in lists but in different order.
-            # Compare on sets since they do not need any order.
-            if set(getattr(alarm, attr)) != set(action):
-                changed = True
-                setattr(alarm, attr, module.params.get(attr))
+                alarm[key] = value
 
         try:
             if changed:
-                connection.create_alarm(alarm)
-        except BotoServerError as e:
+                connection.put_metric_alarm(AlarmName=alarm['AlarmName'],
+                                            MetricName=alarm['MetricName'],
+                                            Namespace=alarm['Namespace'],
+                                            Statistic=alarm['Statistic'],
+                                            ComparisonOperator=alarm['ComparisonOperator'],
+                                            Threshold=alarm['Threshold'],
+                                            Period=alarm['Period'],
+                                            EvaluationPeriods=alarm['EvaluationPeriods'],
+                                            Unit=alarm['Unit'],
+                                            AlarmDescription=alarm['AlarmDescription'],
+                                            Dimensions=alarm['Dimensions'],
+                                            AlarmActions=alarm['AlarmActions'],
+                                            InsufficientDataActions=alarm['InsufficientDataActions'],
+                                            OKActions=alarm['OKActions'],
+                                            TreatMissingData=alarm['TreatMissingData'])
+        except ClientError as e:
```

Comment:
This one was here originally, but would you be able to also update this to use module.fail_json_aws(e)?  The to_native() call that uses does a better job of handling py2/3 type conversions and is all-around preferred.